### PR TITLE
Hw2 dev

### PR DESCRIPTION
**Pull request by whhsu2**
hw2

### CODE REVIEW

**Comment by lindseysbrown** [comment]
Problem 1: All screenshots included 15/15

**Comment by feiyu-chen96** [comment]
P7 Good job!

File: homeworks/HW2/HW2-final/P5b.py [hunk 1]
```diff
@@ -0,0 +1,22 @@
...
+	print("Initial balance: {}".format(init_balance))
+	def take_money(amount):
+		try:
+			amount = int(amount)
+			if amount > init_balance:
+				raise ValueError
+			else:
+				init_balance = init_balance - amount
+				print("Withdrew {}, new balance = {}".format(amount, init_balance))
+				return init_balance - amount
```
Comment (daninge98) [inline homeworks/HW2/HW2-final/P5b.py]: This return statement should just be `return init_balance`

File: homeworks/HW2/HW2-final/P5b.py [hunk 2]
```diff
@@ -0,0 +1,22 @@
+def make_withdrawal(init_balance):
+	print("The following code does not work, because if we try to change initial balance in the inner function," \
+		"the variable hasn't been defined in that block yet. ")
```
Comment (daninge98) [inline homeworks/HW2/HW2-final/P5b.py]: If this is the case, how do we access the value of `init_balance` in P5a without any errors?

File: homeworks/HW2/HW2-final/P5b.py (hunk 1 above)
Comment (daninge98) [inline homeworks/HW2/HW2-final/P5b.py]: This should be `return init_balance`

File: homeworks/HW2/HW2-final/P5c.py [hunk 3]
```diff
@@ -0,0 +1,21 @@
...
+	def take_money(amount):
+		try:
+			nonlocal init_balance
+			amount = int(amount)
+			if amount > init_balance:
+				raise ValueError
+			else:
+				init_balance = init_balance - amount
+				print("Withdrew {}, new balance = {}".format(amount, init_balance))
+				return init_balance - amount
```
Comment (daninge98) [inline homeworks/HW2/HW2-final/P5c.py]: This should be `return init_balance`. Incorrect demo [-2]

File: homeworks/HW2/HW2-final/P6.py [hunk 4]
```diff
@@ -0,0 +1,40 @@
...
+def np_ave(area):
+	mean_area = np.mean(area)
+
+if __name__ == "__main__":
+	# open circle file and input each radius into list
+	f= open("circles.txt","r")
+	f = f.readlines()
+	radii = [0.0] * len(f)
+	radii_2 = np.zeros(len(f))
+	for i in range(len(f)):
```
Comment (bhavenp) [inline homeworks/HW2/HW2-final/P6.py]: Use `enumerate` when you are looping over a list and want both the index and item.

**Comment by Michael137** [comment]
problem 3: 10/10

**Comment by adityakaran** [comment]
P4:10/10 - Nice job

**Comment by cj-xin** [comment]
### Problem 2

3/3 pt: Plot is correct
1/2 pt: File not closed after `open(...)`
1/1 pt: Raw data processed correctly
1/1 pt: Correct counting of occurrences
1/1 pt: Result printed as specified
2/2 pt: Function written as specified

**Pull request merged**